sdk/core/azure-core/src/main/java/com/azure/core/util/Context.java

```diff
@@ -21,14 +22,28 @@
  */
 @Immutable
 public class Context {
-    private final ClientLogger logger = new ClientLogger(Context.class);
+    private static final ClientLogger LOGGER = new ClientLogger(Context.class);
 
     // All fields must be immutable.
     //
     /**
      * Signifies that no data needs to be passed to the pipeline.
      */
-    public static final Context NONE = new Context(null, null, null);
+    public static final Context NONE = new Context(null, null, null) {
+        @Override
+        public Optional<Object> getData(Object key) {
```

Comment:
mind give an explanation why it benefits. trying to learn from it.

Reply:
is it because it is static?

Reply:
`getData` is less of an optimization given that an empty `Context` will complete quickly. `getValues` is a larger optimization as it removes the instantiation of a `HashMap` which would never contain values.